pipeline/transformer/transformer_calm/src/main/scala/uk/ac/wellcome/platform/transformer/calm/CalmTransformer.scala

```diff
@@ -1,28 +1,174 @@
+      title <- title(record)
+      workType <- workType(record)
+      collection <- collection(record)
+      language <- language(record)
+    } yield
+      WorkData(
+        title = Some(title),
+        otherIdentifiers = otherIdentifiers(record),
+        workType = Some(workType),
+        collection = Some(collection),
+        subjects = subjects(record),
+        language = language,
+        mergeCandidates = mergeCandidates(record),
+        items = items(record, accessStatus),
+        contributors = contributors(record),
+        physicalDescription = physicalDescription(record),
+        notes = notes(record)
+      )
+
+  def sourceIdentifier(record: CalmRecord): SourceIdentifier =
+    SourceIdentifier(
+      value = record.id,
+      identifierType = CalmIdentifierTypes.recordId,
+    )
+
+  def otherIdentifiers(record: CalmRecord): List[SourceIdentifier] =
+    identifierMapping.toList.flatMap {
+      case (key, idType) =>
+        record
+          .getList(key)
+          .map(id => SourceIdentifier(identifierType = idType, value = id))
+    }
+
+  def mergeCandidates(record: CalmRecord): List[MergeCandidate] =
+    record
+      .get("BNumber")
+      .map { id =>
+        MergeCandidate(
+          SourceIdentifier(
+            identifierType = IdentifierType("sierra-system-number"),
+            ontologyType = "Work",
+            value = id
+          )
+        )
+      }
+      .toList
+
+  def title(record: CalmRecord): Result[String] =
+    record
+      .get("Title")
+      .map(Right(_))
+      .getOrElse(Left(new Exception("Title field not found")))
+
+  def workType(record: CalmRecord): Result[WorkType] =
+    record
+      .get("Level")
+      .map {
+        case "Collection" => Right(WorkType.ArchiveCollection)
+        case "Section"    => Right(WorkType.ArchiveSection)
+        case "Series"     => Right(WorkType.ArchiveSeries)
+        case "SubSeries"  => Right(WorkType.ArchiveSubSeries)
+        case "Item"       => Right(WorkType.ArchiveItem)
+        case level        => Left(new Exception(s"Unrecognised work type: $level"))
+      }
+      .getOrElse(Left(new Exception("Level field not found.")))
+
+  def collection(record: CalmRecord): Result[Collection] =
+    record
+      .get("RefNo")
+      .map { path =>
+        Right(Collection(label = record.get("AltRefNo"), path = path))
+      }
+      .getOrElse(Left(new Exception("RefNo field not found")))
+
+  def items(record: CalmRecord,
+            status: Option[AccessStatus]): List[Item[Unminted]] =
+    List(
+      Item(
+        title = None,
+        locations = List(physicalLocation(record, status))
       )
     )
+
+  def physicalLocation(record: CalmRecord,
+                       status: Option[AccessStatus]): PhysicalLocation =
+    PhysicalLocation(
+      locationType = LocationType("scmac"),
+      label = "Closed stores Arch. & MSS",
+      accessConditions = List(accessCondition(record, status))
+    )
```

Comment:
Is this right?

Reply:
yes, according to @jtweed

Reply:
Any chance of whacking a comment in here for our future selves about this being a temp measure so whern I see it in the future I don't question why it's like this?

As far as I know it's because we're working on locations so this is OK for now.

Reply:
Is it a temporary measure, I'm not sure :thinking: